modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -1009,13 +982,20 @@ else if (StringUtil.equalsIgnoreCase(
 
 		ServiceContext serviceContext = _createServiceContext(userAccount);
 
-		user = _userService.updateExternalReferenceCode(
+		_userService.updateExternalReferenceCode(
 			userAccountId,
 			GetterUtil.getString(
 				userAccount.getExternalReferenceCode(),
 				user.getExternalReferenceCode()));
 
-		_userService.updateStatus(
+		Integer workflowStatus = _getWorkflowStatus(
+			userAccount.getStatusAsString());
+
+		if (workflowStatus == null) {
+			throw new BadRequestException("Status is invalid");
+		}
+
+		user = _userService.updateStatus(
```

Comment:
Can we move this somewhere on top of the method (like in the previous position)? In this way we avoid doing all the stuff if the status is invalid

Reply:
Or better move only the validation on top and the updateStatus call here. What do you think?